### ISSUE TITLE
length / size / Length / Count are in both lookup tables, so one line scores twice

### ISSUE THREAD

**Issue by CognitiveDecisions** [rsionnach/sloppylint#29]
`length`, `size`, `Length` and `Count` appear in both `HALLUCINATED_METHODS`:

https://github.com/rsionnach/sloppylint/blob/e0e3ced/src/sloppy/analyzers/import_validator.py#L238

and `HallucinatedAttribute.INVALID_ATTRIBUTES`:

https://github.com/rsionnach/sloppylint/blob/e0e3ced/src/sloppy/patterns/hallucinations.py#L491

`visit_Call` and `visit_Attribute` both fire on `xs.length()`, since the `Attribute` is the `Call`'s `func`.

### Repro

```
$ printf 'def f(xs):\n    return xs.length()\n' > d.py
$ sloppylint d.py --format compact
  d.py:2  hallucinated_method: 'length' is not a Python method. Use len(list) not list.length
  d.py:2  hallucinated_attribute: Use len(obj) not obj.length - JavaScript pattern
```

Two high-severity findings, 30 score points, one line of code.

### The two tables also disagree

`size` is marked valid in one and invalid in the other:

```python
# import_validator.py:351
"size": (None, None),        # numpy/pandas .size attribute

# hallucinations.py:494
"size": "Use len(obj) not obj.size - JavaScript pattern (unless pandas/set)",
```

The attribute table wins, because `visit_Attribute` fires regardless. So `df.size` on a pandas DataFrame is reported at high severity by a message that names pandas as the exception.

### Suggested fix

`HallucinatedAttribute` could skip an `Attribute` that is the `func` of a `Call` (the comment at hallucinations.py:516 notes this was intended but not done). Tracking the parent during the walk makes it a one-line check, and it also resolves the disagreement by giving each name exactly one owner.

Found on 0.5.1 (`e0e3ced`).
